lib/controllers/login.js

```diff
@@ -50,6 +51,19 @@ module.exports = function(req, res) {
   res.locals.status = req.query.status;
 
   if (req.method === 'POST' && accepts === 'json') {
+    if (req.body.providerData) {
```

Comment:
How about adding `var providerData = req.body.providerData;` then `if (providerData) {`, etc, and using that instead of repeating it?